### PR TITLE
[CI] Add unit test workflows for Windows

### PR DESCRIPTION
**Pull request by yohann-benchetrit**
## Description

- Add `unittest_windows_optdepts_cpu` workflow for `python_version=3.7, 3.8, 3.9, 3.10` and 
- Add `unittest_windows_optdepts_gpu` workflow for `python_version=3.7, 3.8, 3.9, 3.10` and `cuda=116` (workflows are not run for `cuda=117` [as done in the Torchvision repo](https://github.com/pytorch/vision/blob/35f68a09f94b2d7afb3f6adc2ba850216413f28e/.circleci/config.yml#L1496))

**Some unit tests are set to be skipped on Windows**:
- `test/test_rb_distributed.py::test_funcs`: `torch.distributed.rpc` support is unstable for Windows
- `test/test_rb.py:: TestPrototypeBuffers::{test_add,test_extend,test_sample,test_index}` with `rb_type= RemoteTensorDictReplayBuffer`: `torch.distributed.rpc` support is unstable for Windows 
- `test_timeit.py` is taking more time than allowed on Windows:
```Traceback (most recent call last):
  File "C:\Users\circleci\project\test\test_helpers.py", line 935, in test_timeit
    assert abs(val1[0] - w1) < 1e-2
AssertionError: assert 0.01548680295944214 < 0.01
 +  where 0.01548680295944214 = abs((0.015586802959442139 - 0.0001))
```
- `test_shared_.py::test_memmap` : Call to `td.to(SavedTensorDict)` is failing  (for all combinations of `dtype` and `idx`) with traceback:
```
Traceback (most recent call last):
  File "C:\Users\circleci\project\test\test_shared.py", line 178, in test_memmap
    td_saved = td.to(SavedTensorDict)
  File "C:\Users\circleci\project\env\lib\site-packages\tensordict\tensordict.py", line 2796, in to
    td = dest(
  File "C:\Users\circleci\project\env\lib\site-packages\tensordict\tensordict.py", line 4769, in __init__
    self._save(source)
  File "C:\Users\circleci\project\env\lib\site-packages\tensordict\tensordict.py", line 4779, in _save
    torch.save(tensordict, self.filename)
  File "C:\Users\circleci\project\env\lib\site-packages\torch\serialization.py", line 442, in save
    with _open_zipfile_writer(f) as opened_zipfile:
  File "C:\Users\circleci\project\env\lib\site-packages\torch\serialization.py", line 317, in _open_zipfile_writer
    return container(name_or_buffer)
  File "C:\Users\circleci\project\env\lib\site-packages\torch\serialization.py", line 290, in __init__
    super().__init__(torch._C.PyTorchFileWriter(str(name)))
RuntimeError: File C:\Users\circleci\AppData\Local\Temp\tmp8o43ldz4 cannot be opened.
```

- `test_collector.py::test_output_device_consistency`: `Windows Access Violations` and `BrokenPipeError` for certain combinations of python/policy_device/device/num_env
- `test_collector.py::test_collector_batch_size`: Test timeout (> 10 min) on CI pipeline Windows machine with GPU for `num_env=3`
- `test_collector.py::test_collector_device_combinations`: `Access violation` in `torch.storage`

## Motivation and Context

TorchRL is currently built and tested for OsX and Linux only and there is some demand for a windows built. 
This requires adding some test workflows for Windows.

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).

### CODE REVIEW

File: torchrl/data/tensor_specs.py [hunk 1]
```diff
@@ -469,7 +469,7 @@ def encode(
         if (val >= space.n).any():
             raise AssertionError("Value must be less than action space.")
 
-        val = torch.nn.functional.one_hot(val, space.n).to(torch.long)
+        val = torch.nn.functional.one_hot(val.long(), space.n).to(torch.long)
```
Comment (yohann-benchetrit) [inline torchrl/data/tensor_specs.py:472]: When running `test_mult_discrete_action_spec_reconstruct`, this explicit cast was required for `one_hot` to accept its input (line 993 - `actions_tensors = [action_spec.encode(a) for a in actions_numpy]`)

Comment (vmoens) [reply torchrl/data/tensor_specs.py:472]: do we need the `to(long)` in the end then?

**Review by vmoens — APPROVED** [review]
LGTM, just a minor comment
Thanks a mil!

File: torchrl/data/tensor_specs.py (hunk 1 above)
Comment (yohann-benchetrit) [reply torchrl/data/tensor_specs.py]: No indeed. Removed

File: test/test_collector.py [hunk 2]
```diff
@@ -141,6 +145,10 @@ def _is_consistent_device_type(
     return tensordict_device_type == passing_device_type
 
 
+@pytest.mark.skipif(
+    _os_is_windows and _python_is_3_10,
+    reason="Windows Access Violation in torch.multiprocessing / BrokenPipeError in multiprocessing.connection",
+)
```
Comment (yohann-benchetrit) [inline test/test_collector.py:151]: Post-acceptance change: skip all `test_output_device_consistency` parameter combinations when using python3.10 as tests keep crashing with new combinations (that were previously passing in previous PR).

**Review by vmoens — APPROVED** [review]

**Pull request merged**